python/ray/util/client/common.py

```diff
@@ -60,7 +62,41 @@ def __del__(self):
 
 
 class ClientObjectRef(ClientBaseRef):
-    pass
+    def __await__(self):
+        return self.as_future().__await__()
+
+    def as_future(self):
+        loop = asyncio.get_event_loop()
+        fut = loop.create_future()
+
+        def set_value(resp):
```

Comment:
Can we just use _on_completed with a simpler callback? So the deserialization code isn't duplicated. Something like
```
def set_result(obj):
    def inner():
        if isinstance(obj, Exception):
            fut.set_exception(obj)
        ...
     loop.call_soon_threadsafe(inner)
    
```

Reply:
Ahh yep, I think so!!